dispatcher/queue_test.go

```diff
@@ -0,0 +1,63 @@
+package dispatcher
+
+import (
+	"math/rand"
+	"strconv"
+	"testing"
+	"time"
+
+	"github.com/newscred/webhook-broker/storage/data"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestPriorityQueueBasic(t *testing.T) {
+	rand.Seed(time.Now().Unix())
+	pq := NewJobPriorityQueue()
+	priorities := rand.Perm(100)
+
+	for _, priority := range priorities {
+		messagePayload := strconv.Itoa(priority)
+		msg, _ := data.NewMessage(channel, producer, messagePayload, "application/json")
+		dj := &data.DeliveryJob{Message: msg}
+		pq.Enqueue(&Job{dj, uint(priority)})
+	}
+	for expectedPriority := 99; expectedPriority >= 0; expectedPriority-- {
+		assert.Equal(t, pq.Len(), expectedPriority+1)
+		topJob := pq.Dequeue()
+		assert.Equal(t, uint(expectedPriority), topJob.Priority)
+		assert.Equal(t, strconv.Itoa(expectedPriority), topJob.Data.Message.Payload)
+	}
+}
+
+func BenchmarkPriorityQueue(b *testing.B) {
+
+	for i := 0; i < b.N; i++ {
+		stressPriorityQueue()
+	}
+}
+
+func stressPriorityQueue() {
+	MaxQueueSize := 100000
```

Comment:
As far as I understand, maximum size of the job queue is decided by [this](https://github.com/newscred/webhook-broker/blob/2b22013271c903ea428e8430c070502f50ae0a15/webhook-broker-dc.cfg#L6) `max-message-queue-size` config value which takes on different value in different configs (e.g: 10000, 100000, [10000000](https://github.com/newscred/webhook-broker/blob/2b22013271c903ea428e8430c070502f50ae0a15/webhook-broker-dc.cfg#L6) )